### PR TITLE
Update winforms migration sample

### PR DESCRIPTION
**Pull request by adegeo**
## Summary

- Ran the .NET Framework version through .NET Upgrade Assistant
- Target .NET 9
- Renamed folder to be generic `net-windows`

Supports https://github.com/dotnet/docs-desktop/issues/1808

### CODE REVIEW

File: windowsforms/matching-game/net-windows/vb/MatchingGame/App.config [hunk 1]
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<configuration>
+    <startup>
+        <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5" />
```
Comment (Tanya-Solyanik) [inline windowsforms/matching-game/net-windows/vb/MatchingGame/App.config:4]: .NET projects don't use this file, please remove it.

File: windowsforms/matching-game/readme.md [hunk 2]
```diff
@@ -1,8 +1,8 @@
 # Matching Game Sample
 
-These two samples are used by the [.NET Upgrade Assistant](https://docs.microsoft.com/dotnet/core/porting/upgrade-assistant-wpf-framework) article.
+These two samples are used by the [.NET Upgrade Assistant](https://docs.microsoft.com/dotnet/core/porting/upgrade-assistant-winforms-framework) article.
```
Comment (Tanya-Solyanik) [inline windowsforms/matching-game/readme.md:3]: Nice!

**Comment by Tanya-Solyanik** [comment]
FYI - the readme file in the root of samples/windowsforms folder says "NET3.1" Is this on your radar to update as well?

**Comment by adegeo** [comment]
@Tanya-Solyanik OK I updated the root file. I rewrote it since it was so old.

**Review by Tanya-Solyanik — APPROVED** [review]

**Pull request merged**